pyccel/ast/core.py

```diff
@@ -2569,6 +2569,17 @@ def body(self):
         """ CodeBlock containing all the statements in the function """
         return self._body
 
+    @body.setter
+    def body(self, body):
+        if iterable(body):
+            body = CodeBlock(body)
+        elif not isinstance(body, CodeBlock):
+            raise TypeError('body must be an iterable or a CodeBlock')
+        self._body.invalidate_node()
```

Comment:
```suggestion
        self._body.remove_user_node(self)
```

@sboof911 This is the line I wrote wrong